packages/manager/apps/web/client/app/hosting/database/order/components/private-database/private-database.controller.js

```diff
@@ -0,0 +1,161 @@
+        catalogItemTypeName: workflowConstants.CATALOG_ITEM_TYPE_NAMES.PLAN,
+        productName: CATALOG_PRODUCT.name,
+        getPlanCode: this.getPlanCode.bind(this),
+        onGetConfiguration: () => this.getConfiguration(),
+      },
+      workflowType: workflowConstants.WORKFLOW_TYPES.ORDER,
+    };
+
+    // select privateSQL by default
+    this.order = {
+      datacenter: this.datacenter,
+      type: 'privateSQL',
+    };
+
+    this.loadingDependencies = true;
+    this.$q
+      .all({
+        services: this.OrderPrivateDatabaseService.getServices(),
+        dependencies: this.loadDependencies(),
+      })
+      .then(({ services }) => {
+        this.services = services;
+      })
+      .catch((error) => {
+        this.handleError(error);
+      })
+      .finally(() => {
+        this.loadingDependencies = false;
+      });
+  }
+
+  loadDependencies() {
+    return this.OrderPrivateDatabaseService.getCatalog(this.user.ovhSubsidiary)
+      .then((catalog) => {
+        this.catalog = catalog;
+        this.productOffers.workflowOptions.catalog = this.catalog;
+        return this.OrderPrivateDatabaseService.getProducts(catalog);
+      })
+      .then((products) => {
+        this.products = products;
+        this.catalogProducts = this.OrderPrivateDatabaseService.constructor.getCatalogProducts(
+          this.catalog,
+          products,
+        );
+        return true;
+      });
+  }
+
+  getVersions(product) {
+    this.catalogProduct = find(this.catalogProducts, {
+      planCode: product.planCode,
+    });
+    const engines = find(this.catalogProduct.configurations, {
+      name: 'engine',
+    });
+
+    if (!engines || !engines.values) {
+      return [];
+    }
+
+    return orderBy(
+      engines.values.sort().map((engine) => {
+        const [engineLabel, engineVersion] = engine.split('_');
+        const name = this.$translate.instant(
+          `hosting_database_order_private_version_${engineLabel}`,
+          { version: engineVersion },
+        );
+
+        return {
+          name,
+          value: engine,
+          engineLabel,
+          engineVersion: parseFloat(engineVersion),
+        };
+      }),
+      ['engineLabel', 'engineVersion'],
+      ['asc', 'desc'],
+    );
+  }
+
+  async updateDatacenter(serviceName) {
+    const datacenter = await this.getDatacenter(serviceName);
+
+    this.$timeout(() => {
+      this.order.datacenter = datacenter;
+    });
+  }
+
+  updateVersionSelector(product) {
+    this.product = product;
+    this.versions = this.getVersions(product);
+  }
+
+  getDatacenter(serviceName) {
+    return this.OrderPrivateDatabaseService.getDatacenter(serviceName);
+  }
+
+  cancel() {
+    if (this.onCancel && isFunction(this.onCancel)) {
+      this.onCancel();
+    }
+  }
+
+  handleSuccess(order) {
+    this.$window.open(order.url, '_blank');
+    if (this.onSuccess && isFunction(this.onSuccess)) {
+      this.onSuccess({ url: order.url });
+    }
```

Comment:
i think we can avoid to use lodash by adding doing this:

```suggestion
    this.$window.open(order.url, '_blank');
    if (this.onSuccess && this.onSuccess instanceof Function) {
      this.onSuccess({ url: order.url });
    }
```

Reply:
Below is the lodash isFunction definition. Its not same as using instanceof

_.isFunction = function(obj) {
  return !!(obj && obj.constructor && obj.call && obj.apply);
};